enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,256 @@
+title: single-node-installation-bootstrap-in-place
+authors:
+  - "@dhellmann"
+  - "@eranco"
+  - "@romfreiman"
+  - "@tsorya"
+reviewers:
+  - TBD
+  - "@mrunalp"
+  - "@markmc"
+  - "@deads2k"
+  - "@wking"
+  - "@eparis"
+  - "@hexfusion"
+approvers:
+  - TBD
+creation-date: 2020-12-13
+last-updated: 2020-12-13
+status: implementable
+see-also:
+  - https://github.com/openshift/enhancements/pull/560
+  - https://github.com/openshift/enhancements/pull/302
+---
+
+# Single Node Production Edge Installer - bootstrap in place
+
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+As we add the new [`single-node production deployment`](https://github.com/openshift/enhancements/pull/560/files)
+we need a way to insall such cluster without extra node dependency for bootstrap.
+
+This enhancement describes the flow for installing Single Node Openshift using liveCD that perform the bootstrap logic and later reboot to become the single node.
+
+## Motivation
+
+Currently, all OpenShift installations use an auxiliary bootstrap node.
+The bootstrap node creates a temporary control plane that is required for launching the actual cluster.
+
+Removing the bootstrap node is a key for the 5G RAN use case and to the more general topic of single-node OCP clusters for edge use cases.
+
+Requiring the axillary node for installing a Single Node Openshift brings to play some requirements:
+1. The obvious additional node, in most cases there isn't additional Hardware/VM at the edge site.
+2. Requires external dependencies:
+   a. Load balancer (only for bootstrap phase)
+   b. Requires DNS (configured per installation)
+3. Cannot use Bare Metal IPI:
+   a. Adds irrelevant dependencies - vips, keepalived, mdns
+   b. Requires same L2 between bootstrap and the Single Node Openshift
+
+### Goals
+
+* This enhancement describes an approach for insallting Single Node OpenShift 
+  for production use.
+* Minimal changes to openshift installer, this implementation shouldn't effect existing deployment flows.
+* Installation should result a clean Single Node Openshift, without bootstrap leftovers.
+* Rely on a livecd, so it would be able to trigger the full installation using virtual media.
```

Comment:
"rely on a liveCD" doesn't seem goal-worthy.  What is it about virtual media that helps?  A liveCD approach is just one implementation that gets you whatever that benefit is.

Reply:
the liveCD approach is what makes it possible to run the bootstrap flow in place while keeping the resulting single node OCP clean from bootstrap leftovers.

Reply:
IIRC, the live ISO environment provides a scratch place to write bootstrapping files so that they don't end up on the real node. This eliminates a potential source of errors and confusion when debugging problems.